onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1209,6 +1209,17 @@ Status TensorrtExecutionProvider::OnRunEnd(bool sync_stream) {
   return Status::OK();
 }
 
+nvinfer1::IBuilder& TensorrtExecutionProvider::GetBuilder() const {
+  if (!builder_) {
+    TensorrtLogger& trt_logger = GetTensorrtLogger();
+    {
+      auto lock = GetApiLock();
+      builder_ = std::unique_ptr<nvinfer1::IBuilder>(nvinfer1::createInferBuilder(trt_logger));
+    }
+  }
+  return *builder_;
```

Comment:
it's a little unusual to return reference?
Couldn't we return raw pointer?
Then many of the lines below wouldn't have to be changed. you could continue to use trt_builder->
also would be consistent with storing the raw pointer in the trt func_state